src/main/java/org/webshar/hrms/controller/ResourcePermissionController.java

```diff
@@ -0,0 +1,125 @@
+package org.webshar.hrms.controller;
+
+import java.util.Collections;
+import java.util.List;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.webshar.hrms.model.db.ResourcePermission;
+import org.webshar.hrms.request.resourcepermission.ResourcePermissionCreateRequest;
+import org.webshar.hrms.request.resourcepermission.ResourcePermissionUpdateRequest;
+import org.webshar.hrms.response.BatchResponse;
+import org.webshar.hrms.response.Response;
+import org.webshar.hrms.service.ResourcePermissionService;
+import org.webshar.hrms.service.exception.EntityAlreadyExistsException;
+import org.webshar.hrms.service.exception.EntityNotFoundException;
+import org.webshar.hrms.service.exception.ServiceException;
+
+@RestController
+@RequestMapping("/")
+public class ResourcePermissionController
+{
+
+  @Autowired
+  ResourcePermissionService resourcePermissionService;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ResourcePermissionController.class);
+
+  @GetMapping(value = "/services/api/web/hrms/resource_permissions/{resource_permission_id}")
```

Comment:
As per REST API practice, hyphen is preferred over underscore for resource path.
e.g. /services/api/web/hrms/resource-permissions instead of /services/api/web/hrms/resource_permissions.
Need to update this and rest of the paths.

Also, I think the below API is more suitable in this case.
GET /services/api/web/hrms/resources/{resource_id}/permissions.
Give a thought about this and implement accordingly.

Reply:
Agree on first point which is use hyphen over underscore.  With regards to 2nd suggestion, we haven't implemented that scenario  where we are fetching permission of given resource. The above implementation is just fetch the  record from  resource_permission table for a give id. While implementing fetch the permission of given recourse will use the above suggestion end point.

Reply:
2nd point is related to one-to-many modeling of role to permission. See another comment. 
Give a thought on that accordingly.